feature/google/src/main/java/com/revenuecat/purchases/google/attribution/GoogleDeviceIdentifiersFetcher.kt

```diff
@@ -45,7 +47,14 @@ class GoogleDeviceIdentifiersFetcher(
         try {
             val adInfo = AdvertisingIdClient.getAdvertisingIdInfo(applicationContext)
             if (!adInfo.isLimitAdTrackingEnabled) {
-                advertisingID = adInfo.id
+                if (adInfo.id == noPermissionAdvertisingIdValue) {
+                    log(
+                        LogIntent.USER,
+                        AttributionStrings.GOOGLE_PLAY_ADVERTISING_ID_NOT_AVAILABLE,
+                    )
+                } else {
+                    advertisingID = adInfo.id
+                }
```

Comment:
Yes, we shouldn't be adding the permission to our SDK, otherwise kids apps for example would face issues.

I think what you did here is what we should be doing